File: expression.py
n, s = map(int, input().split())
numbers = list(map(int, input().split()))

ans = ''

l_numbers = len(numbers)

def exp(k, sum, res):

  if k == l_numbers:
    if sum == s:
      print(res + '=' + str(s))
      exit()
    return

  exp(k + 1, sum + numbers[k], res + '+' + str(numbers[k]))    
  exp(k + 1, sum - numbers[k], res + '-' + str(numbers[k]))


exp(1, numbers[0], str(numbers[0]))


print('No solution')

File: expression.cpp
#include <iostream>

using namespace std;

int s, n;

void print_arr(int arr[], int len, string mark = "", char sep = '\n') {
	for (int i = 0; i < len; i++) 
		cout << arr[i] << mark << sep;
}

void print_char(char arr[], int len, string mark = "", char sep = '\n') {
	for (int i = 0; i < len; i++) 
		cout << arr[i] << mark << sep;
}

void fill_arr(int arr[], int len) {
	for (int i = 0; i < len; i++)
		cin >> arr[i];
}


int exp(int arr[], int idx, char signs[], int res) {

	if (idx == n) {

		if (res == s) {
			
			for (int i = 0; i < n-1; i++) {
				cout << arr[i];
				cout << signs[i];
			}

			cout << arr[n-1];
			cout << "=";
			cout << s;

			exit(0);
		}
		
		return res;
	}

	int el = arr[idx];

	signs[idx-1] = '+';
	exp(arr, idx + 1, signs, res + el);
	signs[idx-1] = '-';
	exp(arr, idx + 1, signs, res - el);

	return 0;

}


int main() {

	cin >> n >> s;

	int nums[n];
	char signs[n-1];	
	
	fill_arr(nums, n);

	exp(nums, 1, signs, nums[0]);

	cout << "No solution";

	return 0;
}
